src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -307,14 +337,29 @@ else if (entry.getValue() instanceof Double)
     @Nonnull
     public OptionData setName(@Nonnull String name)
     {
-        Checks.notEmpty(name, "Name");
-        Checks.notLonger(name, MAX_NAME_LENGTH, "Name");
-        Checks.isLowercase(name, "Name");
-        Checks.matches(name, Checks.ALPHANUMERIC_WITH_DASH, "Name");
+        checkName(name);
         this.name = name;
         return this;
     }
 
+    /**
+     * Sets the name localizations of this option.
+     *
+     * @param  name
+     *         The translated name to put
+     * @param  locales
+     *         The locales to associate the translated name with
+     *
+     * @return This builder instance, for chaining
+     */
+    @Nonnull
+    public OptionData setName(@Nonnull String name, @Nonnull Locale... locales)
+    {
+        //Checks are done in LocalizationMap
+        nameLocalizations.setTranslations(name, locales);
+        return this;
+    }
```

Comment:
this doesn't expose enough of the underlying API to be useful imo. We should probably remove it and indicate that users should use the localization map getters + setters on the map, or we need to find a better way to express chainability for localization here.